### PR TITLE
Allow to change cipher suite

### PR DESCRIPTION
**Pull request by TimSiefert**
Hi,

i have added an optional option to allow to set the used Cipher Suite for the TLS communication (server and client).

This is necessary in my case, because my MQTT devices (which I cant change/update) use and older cipher suite than the default values.

Microsoft changed the default values in .NET 5.0 and that means my devices work with .NET CORE 3.1 but not with .NET 6.0
https://docs.microsoft.com/en-us/dotnet/core/compatibility/cryptography/5.0/default-cipher-suites-for-tls-on-linux

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/MQTTnet?pullRequest=1381) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/MQTTnet?pullRequest=1381) before we can accept your contribution.<br/><br/>:x: TimSiefert [sign now](https://cla.dotnetfoundation.org/dotnet/MQTTnet?pullRequest=1381)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/MQTTnet?pullRequest=1381) it.</sub>

File: Source/MQTTnet/Implementations/MqttTcpServerListener.cs [hunk 1]
```diff
@@ -178,11 +178,24 @@ async Task TryHandleClientConnectionAsync(CrossPlatformSocket clientSocket)
...
+                        await sslStream.AuthenticateAsServerAsync(
+                            new SslServerAuthenticationOptions()
+                            {
+                                ServerCertificate = _tlsCertificate,
+                                ClientCertificateRequired = _tlsOptions.ClientCertificateRequired,
+                                EnabledSslProtocols = _tlsOptions.SslProtocol,
+                                CertificateRevocationCheckMode = _tlsOptions.CheckCertificateRevocation ? X509RevocationMode.Online : X509RevocationMode.NoCheck,
+                                EncryptionPolicy = EncryptionPolicy.RequireEncryption,
+                                CipherSuitesPolicy = _tlsOptions.CipherSuitesPolicy
+                            });
```
Comment (chkr1011) [inline Source/MQTTnet/Implementations/MqttTcpServerListener.cs:191]: Please add .ConfigureAwait...

File: Source/MQTTnet/Client/Options/MqttClientTlsOptions.cs [hunk 2]
```diff
@@ -29,6 +29,8 @@ public sealed class MqttClientTlsOptions
 
 #if NETCOREAPP3_1 || NET5_0_OR_GREATER
         public List<System.Net.Security.SslApplicationProtocol> ApplicationProtocols { get; set; }
+        
+        public System.Net.Security.CipherSuitesPolicy CipherSuitesPolicy { get; set; }
```
Comment (chkr1011) [inline Source/MQTTnet/Client/Options/MqttClientTlsOptions.cs:33]: Is _null_ a proper value for this property? I do not know details about the new API in .net...

**Review by chkr1011 — CHANGES_REQUESTED** [review]
Thank you for your contribution. I have some small change requests.

File: Source/MQTTnet/Implementations/MqttTcpServerListener.cs (hunk 1 above)
Comment (TimSiefert) [reply Source/MQTTnet/Implementations/MqttTcpServerListener.cs]: Oh yes, You are totally right. I have added it.

File: Source/MQTTnet/Client/Options/MqttClientTlsOptions.cs (hunk 2 above)
Comment (TimSiefert) [reply Source/MQTTnet/Client/Options/MqttClientTlsOptions.cs:33]: Yes, it is the default value.
See https://docs.microsoft.com/en-us/dotnet/api/system.net.security.sslserverauthenticationoptions.ciphersuitespolicy?view=net-6.0#system-net-security-sslserverauthenticationoptions-ciphersuitespolicy

> When set to null, the operating system default is used.

**Comment by chkr1011** [comment]
Thanks for your contribution. Good work 👍

**Pull request merged**